Level.py

```diff
@@ -1,15 +1,210 @@
+                          self.level_size[1] / self.map_canvas.map_size[1]]
+
+        # initialise la matrice
+        self.xmax = self.map_canvas.map_size[0]
+        self.ymax = self.map_canvas.map_size[1]
+        self.matrix = [[0 for _ in xrange(self.xmax)] for _ in xrange(self.ymax)]
+
+        # rempli la matrice avec les coordonnées
+        x = self.map_canvas.vectical_padding
+        y = self.map_canvas.window.size[1] - self.map_canvas.horizontal_padding
+
+        for indx in range(self.ymax):
+            for indy in range(self.xmax):
+                self.matrix[indy][indx] = [x, y, x + self.tile_size[0], y - self.tile_size[1]]
+                y -= self.tile_size[1]
+            x += self.tile_size[0]
+            y = self.map_canvas.window.size[1]
+
+    """
+    Tiles methods
+    """
+
+    def get_touch_auth(self, tile_type):
+        """
+        test the current tile to get authorizations
+        :param tile_type: string (key of the texture)
+        :rtype: boolean
+        """
+        if tile_type == "A":
+            return True
+        elif tile_type == "W":
+            self.player_path = []
+            return False
+
+    def get_tile_properties(self):
+        """
+        find the current tile
+        :rtype: string (key of the texture)
+        """
+        for indx in range(self.ymax):
+            for indy in range(self.xmax):
+                horizontal_location = self.matrix[indy][indx][0] <= self.point[0] < self.matrix[indy][indx][2]
+                vertical_location = self.matrix[indy][indx][1] >= self.point[1] > self.matrix[indy][indx][3]
+                if horizontal_location and vertical_location:
+                    tile_type = self.map_canvas.map_matrix[indy][indx]['type']
+                    self.player_path.append([indx, indy])
+                    return tile_type
+
+    """"
+    win methods
+    """""
+
+    def get_win_conditions(self):
+        """
+        get wins path and conditions
+        :type: void
+        """
+        for indx in range(self.ymax):
+            for indy in range(self.xmax):
+                if self.map_canvas.map_matrix[indy][indx]['type'] == 'A':
+                    self.win_path.append([indx, indy])
+
+    def test_win_conditions(self):
+        """
+        test if players win
+        :rtype: boolean
+        """
+        for entry in self.win_path:
+            if entry not in self.player_path:
+                return False
+        return True
+
+    """"
+    Touch methods
+    """""
+
+    def on_touch_down(self, touch):
+
+        # sauvegarde le touch
+        ud = touch.ud
+        ud['identifier'] = str(touch.uid)
+        self.point = [touch.x, touch.y]
+        self.player_path = []
+
+        # dessine le point si l'emplacement est valide sinon supprime le dessin.
+        tile_type = self.get_tile_properties()
+        if self.get_touch_auth(tile_type):
+            with self.canvas:
```

Comment:
I don't really like `with` I find that not very readable.
I prefer pushing : see MapCanvas